rules/framework/vfs_overlay.bzl

```diff
@@ -36,31 +36,82 @@ def _find_top_swiftmodule_file(swiftmodules):
             return file
     return None
 
+# Builds out the VFS subtrees for a given set of paths. This is useful to
+# construct an in-memory rep of a tree on the file system
+def _build_subtrees(paths, vfs_prefix):
+    # It uses a O(NVFSParts) sized helper dict to avoid O(NPathComponents^2)
+    # worst case runtime
+    subdirs_json = {"contents": [], "type": "directory", "name": "root"}
+    subdirs = struct(dict = {}, json = subdirs_json)
+    for path_info in paths:
+        path = path_info.framework_path
+
+        parts = path.split("/")
+
+        # current pointer to the current subdirs while walking the path
+        curr_subdirs = subdirs
+
+        # Loop the _framework_ path and add each dir to the current tree.
+        # Assume the last bit is a file then add it as a file
+        for part in parts:
+            if idx == len(parts) - 1:
+                ext_c = _get_external_contents(vfs_prefix, path_info.path)
+                curr_subdirs.dict[part] = -1
+                curr_subdirs.json["contents"].append({"name": part, "type": "file", "external-contents": ext_c})
+                break
+
+            # Lookup a value for the current subdirs, otherwise append
+            next_subdirs = curr_subdirs.dict.get(part, None)
```

Comment:
I might misunderstand the logic, but looks like curr_subdirs may be None here if len(parts) > 1.

curr_subdirs is initialized as subdirs, which is struct(dict = {}, json = subdirs_json).
next_subdirs will be None if len(parts) > 1.
Then curr_subdirs will be set None on line 72.

Reply:
`curr_subdirs` isn't set to non-None value for what I could tell. If you trace the execution, you'll likely find this invariant to be held true. Mainly, `next_subdirs` is always set to a non-None value. If you have an example which invalidates the invariant I'd be curious to see.

Reply:
You're right, somehow I missed the `not`